### PR TITLE
Pass Documentation up to LSP and add "rust" to our codeblocks there

### LINKED ISSUE #699: Documentation all the way to the LSP
One thing that I would like to do is make `CompletionItem` bring `Documentation` all the way up to the LSP and do the same for `CallInfo` and possibly `hover`... basically anything that renders documentation. Then in the LSP we can add a conversion from our `Documentation` to the LSP's `Documentation` that in theory could choose the type based on what's supported and would also do the markdown massaging from https://github.com/rust-analyzer/rust-analyzer/blob/ffcf61884245ca515f08f8685dc324b9db727e53/crates/ra_ide_api/src/call_info.rs#L131-L147

### PR DESCRIPTION
**Pull request by kjeremy**
Fixes #699

### CODE REVIEW

File: crates/ra_lsp_server/src/markdown.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    // Massage markdown
+    let mut processed_lines = Vec::new();
+    let mut in_code_block = false;
+    for line in docs.lines() {
+        if line.starts_with("```") {
+            in_code_block = !in_code_block;
+        }
+
+        let line = if in_code_block && line.starts_with("```") && !line.contains("rust") {
+            "```rust".into()
```
Comment (bjorn3) [inline crates/ra_lsp_server/src/markdown.rs]: This will turn ```` ```plain ```` into ```` ```rust ````.

Comment (matklad) [reply crates/ra_lsp_server/src/markdown.rs]: I woudn't be against pulling-in a markdown parser...

Comment (kjeremy) [reply crates/ra_lsp_server/src/markdown.rs]: Agreed. I think ultimately we could do things like: text only clients get stripped out "```" for readability, and we could run doc tests with a code-lens etc. We'll probably also need one to make links work in markdown.

**Pull request merged**